teaspoons-cli/tests/logic/test_pipelines_logic.py

```diff
@@ -0,0 +1,70 @@
+# tests/logic/test_pipelines_logic.py
+
+from unittest.mock import patch, Mock
```

Comment:
You might like [mockito](https://pypi.org/project/mockito/) if you're unfamiliar with it.  I'm not sure if that would make some of the mocking stuff easier.

Reply:
oh thanks, that's a much nicer interface!